### PR TITLE
Misc updates to make it run with latest version of keras and support for conversion to tensorflow frozen model

### PR DESCRIPTION
**Pull request by ksachdeva**
Hi @shamangary 

As I mentioned earlier, usage of Lambda layers create issues to do conversions and require python as the run time. My main goal was to be able to convert the models to tensorflow. 

Fortunately I managed to write the custom layers and conversion process in such a way that **your pre-trained models can still be used & imported**. This also helps in making sure that I am not introducing the bugs. I have ran the test script and validated that there is no change in the accuracy of various models. 

## Difficult part:

To support latest keras, I had to import the batch_dot from previous versions. Capsule layer uses batch_dot which in latest keras complains. Now I do not know Capsule networks in depth so for now I have place the batch_dot implementation along side. The correct way to do it would be to use a new implemention of Capsule network. I indeed tried it but the way the new implementation (from keras-contrib) works is differen from what you have. Usage of new capsule implementation will lead to not be able to use your pretrained models therefore I added the compliant batch_dot implementation. 

## New feature thanks to custom layers
Finally there is a script in training_and_testing folder that is now capable of converting to tensroflow pb model. From this point conversion to tensorRT, onnx etc should be simple. 

## Potential merge issue
When I started the branch I did not see your revision where you changed 8*8 to self.map_x_y*self.map_x_y. I have added that change however github is complaining that the pull request can not be merged. I hope it is not a problem.

Let me know if you have questions. 

Regards
Kapil

### CODE REVIEW

**Comment by shamangary** [comment]
Hello @ksachdeva, the code looks great. However, I am in the middle of something, it may take a while for me to review the code and the capsule thing. I have struggled for transferring the keras model to tf as well. Big thanks to you.
